documentation/decisions/adr/2023-12-15-deliverable-reporting-strategy.md

```diff
@@ -0,0 +1,235 @@
+
+
+- **Pros**
+  - Ensures that there is just one source of truth for the list of issues associated with a given 30k deliverable.
+  - Supports filtering both within the repository and within GitHub projects (e.g. roadmap and sprint board).
+  - Supports grouping both within the GitHub projects and within GitHub insight reporting.
+- **Cons**
+  - Prevents the engineering team from using GitHub milestones to organize and group related issues outside of the 30k deliverable framework.
+  - Milestones that represent 30k deliverables will have a *lot* of tickets unless we work on downscoping 30k deliverables.
+  - Does not support customizing the order of the groups on the sprint board when grouping by milestone -- milestones can only be sorted alphabetically.
+
+### Label
+
+This option involves creating a label for each 30k deliverable with a consistent prefix (e.g. `30k: Public launch`) and then applying this label to each of the issues required for delivery of that 30k.
+
+> [!TIP]
+> **Bottom line:** This is the best option if:
+> - we want a consistent way to filter issues by deliverable in the repository and across GitHub projects
+> - but can compromise on being able to **group** by deliverable in GitHub projects or reporting
+
+- **Pros**
+  - Allows the engineering team to continue to use GitHub milestones to organize issues into units of work that make sense to them.
+  - Supports filtering both within the repository and within GitHub projects (e.g. roadmap and sprint board).
+- **Cons**
+  - Does not easily support grouping in GitHub projects or GitHub insights reporting.
+  - Will result in a large number of labels if 30k labels aren't deleted once a 30k deliverable is complete.
+
+### Deliverable column
+
+This option involves creating a single select "deliverable" column in the GitHub projects, with values for each 30k deliverable, and then selecting the corresponding "deliverable" value for each issue added to the project.
+
+> [!TIP]
+> **Bottom line:** This is the best option if:
+> - we want a consistent way of filtering, grouping, and sorting issues by deliverable in GitHub projects and reporting
+> - but can compromise on being able to filter by deliverable in the **repository**
+
+- **Pros**
+  - Allows the engineering team to continue to use GitHub milestones to organize issues into units of work that make sense to them.
+  - Supports filtering within GitHub projects (e.g. roadmap and sprint board).
+  - Supports grouping within GitHub projects and within GitHub insight reporting.
+  - Supports customizing the order of groups on the sprint board when grouping by deliverable -- for example we can make "Static site public launch" appear above "GET Opportunities" even though GET opportunities would be first alphabetically.
+- **Cons**
+  - Does not support filtering or searching within the GitHub repository.
+  - Will result in a large number of values for the deliverable column if not regularly maintained.
```

Comment:
Yeah, this brings up an added downside: if we're deleting the values from the deliverable column after deliverables have reached a certain point (e.g., they've been 100% completed for over a month, or whatever), then this also means we're deleting the history that they were associated with a certain deliverable, right? 

So our historical reporting would be impacted. Not the biggest deal, but worth noting.

Reply:
Added this note in this commit: [feat: Adds some other cons to the reporting options](https://github.com/HHS/simpler-grants-gov/pull/926/commits/4ecc6ff4d7c1a2c549a3b70d92019c8d4e852733)